### PR TITLE
Unpack CountBucket to give mean

### PR DESCRIPTION
**Pull request by phajy**
This pull request unpacks CountBucket to give mean. The changes include fixing the order of the variables in the list comprehension.

### CODE REVIEW

File: src/valuebucket.jl [hunk 1]
```diff
@@ -90,7 +90,7 @@ end
 
 Base.size(b::CountBucket) = size(b.output)
 function unpack_bucket(b::CountBucket)
-    [n > 0 ? o / n : 0 for (n, o) in zip(b.output, b.ncounts)]
+    [n > 0 ? o / n : 0 for (o, n) in zip(b.output, b.ncounts)]
```
Comment (fjebaker) [inline src/valuebucket.jl:93]: Can you add a test that fails on the old version, but passes on this?

Comment (phajy) [reply src/valuebucket.jl:93]: Test added that should catch this.

**Review by fjebaker — APPROVED** [review]
LGTM

**Comment by fjebaker** [comment]
Thanks Andy!

File: test/runtests.jl [hunk 2]
```diff
@@ -195,6 +194,18 @@ end
         58201.07894736842,
     ]
     y_binned = bucket(X, y, bins; reduction = mean)
+    @test y_binned = [0.00019574174576036845, 0, 0, 0, 0]
```
Comment (fjebaker) [inline test/runtests.jl]: This has a floating point comparison error on the CI: try using the `\approx` equals and adding an `atol` or `rtol` parameter to the macro:

```julia
@test y_binned ≈ [0.00019574174576036845, 0, 0, 0, 0] rtol = 1e-3
```

Comment (phajy) [reply test/runtests.jl]: I've gone with `rtol = 1e-3` here and below. I'm not sure how `rtol` works with zero, but since zero is exact, it should be OK.

**Pull request merged**